protocols/relay/src/priv_client.rs

```diff
@@ -100,12 +104,10 @@ pub struct Client {
     queued_actions: VecDeque<Event>,
 }
 
-impl Client {
-    pub fn new_transport_and_behaviour(
-        local_peer_id: PeerId,
-    ) -> (transport::ClientTransport, Self) {
-        let (transport, from_transport) = transport::ClientTransport::new();
-        let behaviour = Client {
+impl Behaviour {
+    pub fn new_transport_and_behaviour(local_peer_id: PeerId) -> (transport::Transport, Self) {
```

Comment:
Constructors that return a tuple are always a bit odd to interact with. If we look at the ecosystem (channel libraries f.e.), we can see that these are often handled slightly differently. In particular, creating channels happens via free-functions. For example:

```rust
let (sender, receiver) = futures::channel::mpsc::bounded(1);
```

We could do the same thing here:

```rust
let (behaviour, transport) = libp2p::relay::client::new(peer_id);
```

Reply:
yeah, agreed. Updated!